nano/node/bootstrap/bootstrap_ascending.cpp

```diff
@@ -678,12 +717,23 @@ bool nano::bootstrap_ascending::run_one ()
 	return success;
 }
 
+void nano::bootstrap_ascending::throttle_if_needed ()
+{
+	if (!iterator.warmup () && throttle.throttled ())
```

Comment:
Doesn't 'throttle' object access need a lock?

Reply:
As it's written it assumes synchronisation is handled externally.
This seemed appropriate since it's a 1-to-1 association between bootstrap_ascending instance and throttle instance.

Reply:
Why then are you locking accesses of this class in other places, I believe your explanation is not correct and it is in fact being used from multiple threads.

Reply:
In other words, just move the lock acquired on line 725 above this if statement.